### PR TITLE
[Lang] Initial matrix argument support for ti.kernel

### PR DESCRIPTION
**Pull request by mzmzm**
Related issue = #3904

### CODE REVIEW

**Comment by strongoier** [comment]
Welcome to the Taichi community! I suggest reading our [contribution guide](https://docs.taichi.graphics/lang/articles/contribution/contributor_guide). The current step is to solve code format & style issues :-)

**Comment by mzmzm** [comment]
Thanks! I have fixed my code format error.

File: python/taichi/lang/kernel_arguments.py [hunk 1]
```diff
@@ -40,6 +41,14 @@ def decl_scalar_arg(dtype):
     return Expr(_ti_core.make_arg_load_expr(arg_id, dtype))
 
 
+def decl_matrix_arg(matrixtype):
+    tmp = Matrix(n=matrixtype.n, m=matrixtype.m)
+    for i in range(matrixtype.n):
+        for j in range(matrixtype.m):
+            tmp.set_entry(i, j, decl_scalar_arg(matrixtype.dtype))
+    return tmp
```
Comment (strongoier) [inline python/taichi/lang/kernel_arguments.py:49]: This way of constructing `Matrix` will be deprecated soon :-) 
```suggestion
    return Matrix([[decl_scalar_arg(matrixtype.dtype) for _ in matrixtype.m] for _ in matrixtype.n])
```

**Review by strongoier — CHANGES_REQUESTED** [review]
Great work! Only some nits.

File: tests/python/test_matrix_arg.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+
+#Matrix test
+@ti.test()
+def test_matrix_arg():
+    mat1 = ti.Matrix([[1, 2, 3], [4, 5, 6]])
+
+    @ti.kernel
+    def foo(mat: ti.types.matrix(2, 3, ti.i32)) -> ti.i32:
+        assert mat1[0, 0] == 1
```
[the remark this answers is not in the document]
Comment (mzmzm) [reply tests/python/test_matrix_arg.py:12]: 1. Yes. It's a typo here. 
2. I will change it soon.

**Review by strongoier — APPROVED** [review]
LGTM!

**Pull request merged**